Src/VimCore/ModeLineInterpreter.fs

```diff
@@ -62,19 +72,45 @@ type internal ModeLineInterpreter
         // Whether we should allow the setting
         let shouldAllowSetting settingName =
 
-            // For security reasons, we disallow certain local settings.
+            // For security reasons, we disallow certain settings.
             let globalSetting = _globalSettings.GetSetting settingName
             let localSetting = _localSettings.GetSetting settingName
+            let windowSetting = windowSettings.GetSetting settingName
             if Option.isSome globalSetting then
 
                 // Disallow all global settings.
                 false
+
             elif Option.isSome localSetting then
 
                 // Allow any local setting that isn't insecure.
                 _insecureLocalSettingNames
                 |> Seq.contains localSetting.Value.Name
                 |> not
+
+            elif Option.isSome  windowSetting then
+
+                // Allow any window setting that isn't insecure.
+                _insecureWindowSettingNames
+                |> Seq.contains windowSetting.Value.Name
+                |> not
+
+            else
+                false
+
+        // Set the setting if applicable
+        let setSetting settingName settingValue =
+            if _localSettings.GetSetting settingName |> Option.isSome then
+
+                // Only set local settings the first time the buffer is
+                // checked.

```

Comment:
What is the expected behavior when multiple `ITextView` are opened on a single `ITextBuffer`. Presently there is a 1:1 relationship between `ModeLineInterpretter` and `ITextView` meaning if we open multiple views we will interpret the mode line mulitple times. My reading of the comment indicates the opposite is meant to be true.

Reply:
No, there is a one-to-one relationship between `ModeLineInterpreter` and `IVimTextBuffer` which has a one-to-one relationship with `ITextBuffer`. This is what the member function accepts window settings as an argument. The window settings apply to the text view, the local settings apply to the text buffer.

Reply:
But your point is well taken that it would be possible to use the other approach. Conceptually a text buffer can exist without a text view but a text view cannot exist without a text buffer so I chose to load loading settings from the buffer.

Reply:
In any case, there is a detailed unit test that verifies the correct behavior.